source/common/common/key_value_store_base.cc

```diff
@@ -72,6 +83,9 @@ void KeyValueStoreBase::addOrUpdate(absl::string_view key_view, absl::string_vie
   if (max_entries_ && store_.size() > max_entries_) {
     store_.pop_front();
   }
+  if (ttl)
+    ttl_manager.add(ttl.value(), key);
```

Comment:
nit: Braces are required for all control statements include single line if, while, etc. statements.

https://github.com/envoyproxy/envoy/blob/main/STYLE.md